lib/rex/proto/mssql/client.rb

```diff
@@ -116,9 +116,11 @@ def detect_platform_and_arch
           result = {}
 
           server_vars = query('select @@version')[:rows][0][0]
+          arch = server_vars.match(/\b\d+\.\d+\.\d+\.\d+\s\((?<arch>[^)]*)\)/)[:arch] || server_vars
```

Comment:
This is slightly different to the code I paste to yourself over slack, and it'll crash if it doesn't match:

```
3.1.4 :002 > "".match(/\b\d+\.\d+\.\d+\.\d+\s\((?<arch>[^)]*)\)/)[:arch]
(irb):2:in `<main>': undefined method `[]' for nil:NilClass (NoMethodError)
	from /Users/user/.rvm/gems/ruby-3.1.4@pro/gems/irb-1.7.4/exe/irb:9:in `<top (required)>'
	from /Users/user/.rvm/gems/ruby-3.1.4@pro/bin/irb:25:in `load'
	from /Users/user/.rvm/gems/ruby-3.1.4@pro/bin/irb:25:in `<main>'
	from /Users/user/.rvm/gems/ruby-3.1.4@pro/bin/ruby_executable_hooks:22:in `eval'
	from /Users/user/.rvm/gems/ruby-3.1.4@pro/bin/ruby_executable_hooks:22:in `<main>'
```